### PR TITLE
Fix screenshot tests failing

### PR DESCRIPTION
**Pull request by niklasberglund**
This PR addresses screenshot tests failing, especially after the staging PR was merged to main but they were also failing prior to that.

* Removed usage of partner API client for creating accounts
* Issue with ellipsis button again. There's a race condition even though there shouldn't be. Added a wait for the ellipsis button to show before tapping it.
* Removed `MULLVAD_ACCOUNT_TOKEN` which is no longer used
* Made screenshot tests workflow store Xcode test report as artefact,similar to end to end tests
* Specified simulator iOS version. Fastlane was defaulting to 17.0 without specifying a version, and the runner image have very few 17.0 simulators.
* Not producing screenshots for iPad as the tests do not support the iPad UI

- - -
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6336)

### CODE REVIEW

**Review by niklasberglund — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6336)* status: 0 of 6 files reviewed, 1 unresolved discussion (waiting on @buggmagnet and @rablador)

___
*[`ios/Snapfile` line 18 at r1](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6336#-O-1tW4cAFJh4V0XmKxK:-O-5FjvV6h4iQNKHv8sT:b-z84lbr) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/1081af436b7c5fa5e112cd315d8af1fbd9210ce2/ios/Snapfile#L18)):*
<details><summary><i>Previously, buggmagnet wrote…</i></summary><blockquote>

I don't think this change should be here, please revert it.
</blockquote></details>

Unfortunately I think it should be. Agree it looks odd and we might want to do it in a better way, but `ios-screenshot-tests.yml` is using the `MullvadVPN` scheme since before also:

```
xcodebuild \
            -project MullvadVPN.xcodeproj \
            -scheme MullvadVPN \
            -testPlan MullvadVPNScreenshots \
```

I think it would be great if the `MullvadVPNScreenshots` test plan was moved to `MullvadVPNUITests` target, but that would require some other changes because with it under `MullvadVPN` we get screenshots using bundled production data yet using staging environment.

**Comment by niklasberglund** [comment]
Closing this PR for a while, going to iterate on it a bit then reopening

**Review by rablador — APPROVED** [review]
:lgtm:

> Reviewed 3 of 6 files at r1, 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6336#-:-O-KfmGj0nna8wxrB661:bnfp4nl)* status: 0 of 9 files reviewed, 1 unresolved discussion (waiting on @buggmagnet, @mojganii, and @rablador)

**Review by buggmagnet — CHANGES_REQUESTED** [review]
> Reviewed 5 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6336)* status: 0 of 9 files reviewed, 1 unresolved discussion (waiting on @mojganii, @niklasberglund, and @rablador)

___
*[`ios/Snapfile` line 8 at r3](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6336#-O-jpIax5qN2sywfLWGw:-O-jpIax5qN2sywfLWGx:bf9m0aq) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/92af5bd0351dbf7d3906e815538ad98e9430f229/ios/Snapfile#L8)):*
> ```Ruby
>   "iPhone 15 Pro",
>   "iPhone 15 Pro Max"
>   #"iPad Pro (11-inch) (4th generation)",
> ```

We shouldn't remove iPads from the screenshots, especially now that we are removing the iPad layout support, we will want to update the screenshots when this ships.

**Review by niklasberglund — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6336)* status: 0 of 9 files reviewed, 1 unresolved discussion (waiting on @buggmagnet, @mojganii, and @rablador)

___
*[`ios/Snapfile` line 8 at r3](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6336#-O-jpIax5qN2sywfLWGw:-O-jrK290amRwLcyl1GK:b6a7gjd) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/92af5bd0351dbf7d3906e815538ad98e9430f229/ios/Snapfile#L8)):*
<details><summary><i>Previously, buggmagnet wrote…</i></summary><blockquote>

We shouldn't remove iPads from the screenshots, especially now that we are removing the iPad layout support, we will want to update the screenshots when this ships.
</blockquote></details>

Yes, but currently the screenshot tests do not support iPad in the state it is in `main`. After the iPad PR has been merged it should be updated. How about we create a ticket for uncommenting this line, so it doesn't block this PR?

**Review by buggmagnet — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6336#-:-O-jvSfx8gZ7RCGsFZ0g:bnfp4nl)* status: 0 of 9 files reviewed, 1 unresolved discussion (waiting on @mojganii, @niklasberglund, and @rablador)

___
*[`ios/Snapfile` line 8 at r3](https://reviewable.io/reviews/mullvad/mullvadvpn-app/6336#-O-jpIax5qN2sywfLWGw:-O-jvO7a5VaA7g1j7iwh:boqw9py) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/92af5bd0351dbf7d3906e815538ad98e9430f229/ios/Snapfile#L8)):*
<details><summary><i>Previously, niklasberglund (Niklas Berglund) wrote…</i></summary><blockquote>

Yes, but currently the screenshot tests do not support iPad in the state it is in `main`. After the iPad PR has been merged it should be updated. How about we create a ticket for uncommenting this line, so it doesn't block this PR?
</blockquote></details>

You're right, let's leave that as is for now. :thumbsup:

**Pull request merged**
